File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/HashUpdateFile.java
package com.onevgo.functions;

import java.io.*;
import java.security.MessageDigest;

public class HashUpdateFile {
    public static boolean hashUpdateFile(MessageDigest context, String filename) {
        InputStream inputStream = null;
        try {
            inputStream = new BufferedInputStream(new FileInputStream(new File(filename)));
            byte[] b = new byte[8192];
            int len = -1;
            while ((len = inputStream.read(b)) != -1) {
                context.update(b, 0, len);
            }
            return true;
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            if (inputStream != null) {
                try {
                    inputStream.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
        return false;
    }

    public static void main(String[] args) {

    }
}

File: 01_Language/01_Functions/go/src/functions/hash_update_file.go
package functions

import (
	"bufio"
	"hash"
	"os"
)

func HashUpdateFile(context hash.Hash, filename string) bool {
	file, err := os.Open(filename)
	if err != nil {
		return false
	}
	defer file.Close()

	reader := bufio.NewReader(file)
	b := make([]byte, 8192)
	for {
		n, err := reader.Read(b)
		if err != nil {
			break
		}

		context.Write(b[0:n])
	}
	return true
}
